### PR TITLE
Fix uncaught DOMException error

### PR DESCRIPTION
**Pull request by sdb1228**
This should fix https://github.com/atlassian/react-beautiful-dnd/issues/1751 :)

### CODE REVIEW

**Comment by alexreardon** [comment]
When can this happen?

**Comment by sdb1228** [comment]
Hey @alexreardon!  This happens when the dom is cycled out before pretty dnd can unmount (aka this settimeout get executed)

**Comment by alexreardon** [comment]
When did you run into this? I'm guessing in a testing environment?

**Comment by sdb1228** [comment]
Oh! Sorry didn't understand what you meant. I actually ran into this in my development and staging environment on the app I work on. We use Turbo links and sentry had caught this error when someone was navigating away from a react page that had react-beautiful-dnd on it.

**Comment by alexreardon** [comment]
https://github.com/turbolinks/turbolinks ?

**Comment by alexreardon** [comment]
I am thinking we might want to add a test to ensure that we don't break people doing things like this

**Comment by alexreardon** [comment]
For now I'm trying to understand what can cause the issue to occur

**Comment by sdb1228** [comment]
Yup! thats the library.  I can totally add a test.  I can try to find a way to reproduce it.  I was thinking that you could probably experience this if you used something like react-routter  ( https://github.com/ReactTraining/react-router) potentially, but I think the easiest way to trigger this would be to unmount a root level react component at the same time as rendering a full new dom tree so that the timeout happens after the new page is rendered.  I couldn't think of an easy way to reproduce it without spinning up a full rails app with turbo links and everything, but if we really want me to do that I totally can.  In the mean time ill work on writing some tests. :)

**Comment by alexreardon** [comment]
> I think the easiest way to trigger this would be to unmount a root level react component at the same time as rendering a full new dom tree so that the timeout happens after the new page is rendered

Let's add a test for that as I think it is the underlying technique

**Comment by alexreardon** [comment]
Can you please add a test in `unit/integration`?

**Comment by alexreardon** [comment]
(all the tests are moving over to there - all that folder uses react-testing-library)

**Comment by sdb1228** [comment]
Yup! I can totally do that.

File: src/view/use-announcer/use-announcer.js [hunk 1]
```diff
@@ -42,9 +42,11 @@ export default function useAnnouncer(contextId: ContextId): Announce {
         // unmounting after a timeout to let any announcements
         // during a mount be published
         setTimeout(function remove() {
-          // not clearing the ref as it might have been set by a new effect
-          getBodyElement().removeChild(el);
-
+          // checking if element exists before remove to prevent errors
+          if (getBodyElement().contains(el)) {
```
Comment (alexreardon) [inline src/view/use-announcer/use-announcer.js]: I just checked - this is all good for ie11

https://developer.mozilla.org/en-US/docs/Web/API/Node/contains

**Comment by alexreardon** [comment]
Let me know when you would like me to take a look. Keep to get this in for v13!

**Comment by sdb1228** [comment]
Getting on this this morning! should hopefully have tests up before lunch

**Comment by sdb1228** [comment]
Hey @alexreardon it looks like there is a spec already in `test/unit/view/announcer.spec.js:50` that does exactly what I would need to do just need to modify it a little.  You okay if I just add another spec next to it or would you like me to migrate both tests over to that other directory?

**Comment by sdb1228** [comment]
I pushed up the test and put it in the same file.  It has react testing lib in there already I hope that is okay.  Let me know if you want me to move it.

**Comment by sdb1228** [comment]
And it now has passed all checks.  Let me know if there is anything else you would like me to do!

**Comment by alexreardon** [comment]
I am hoping to release 13.x soon and it would be good to get this in. Any chance you will have an opportunity to look at this soon?

**Comment by sdb1228** [comment]
Thanks for verifying! I can't merge because its a protected branch and I am not authorized to do so.  Never had this happen to me before so let me know what you need me to do on my side @alexreardon

**Comment by sdb1228** [comment]
and I just realized that we are going to be in completely different time zones lol.  Ill stay by my phone to merge this but let me know how I can merge into the branch that is protected.  Ill google around to see if I can figure it out in the mean time.

**Comment by alexreardon** [comment]
> I can't merge

I can do the merging once the PR is ready to go

**Comment by sdb1228** [comment]
>Any chance you will have an opportunity to look at this soon?

Awesome! So maybe I missed it is there some feedback I need to address?

**Comment by alexreardon** [comment]
It looks good, i'll merge it very soon

**Pull request merged**